lib/src/network/parse_live_query.dart

```diff
@@ -1,52 +1,180 @@
+    final String _where = query._buildQuery().replaceAll('where=', '');
+    //Convert where condition to Map
+    Map<String, dynamic> _whereMap = Map<String, dynamic>();
+    if (_where != '') {
+      _whereMap = json.decode(_where);
+    }
+
+    final int requestId = _requestIdGenerator();
+
+    try {
+      _webSocket = await WebSocket.connect(_liveQueryURL);
+
+      if (_webSocket != null && _webSocket.readyState == WebSocket.open) {
+        if (_debug) {
+          print('$_printConstLiveQuery: Socket opened');
+        }
+      } else {
+        if (_debug) {
+          print('$_printConstLiveQuery: Error when connection client');
+          return Future.value(null);
+        }
+      }
+
+      _channel = IOWebSocketChannel(_webSocket);
+      _channel.stream.listen((dynamic message) {
+        if (_debug) {
+          print('$_printConstLiveQuery: Listen: ${message}');
+        }
+
+        final Map<String, dynamic> actionData = jsonDecode(message);
 
-  Future<void> subscribe(String className) async {
-    final WebSocket webSocket = await WebSocket.connect(client.data.liveQueryURL);
-    channel = IOWebSocketChannel(webSocket);
-    channel.sink.add(jsonEncode(connectMessage));
-    final Map<String, dynamic> classNameMap = subscribeMessage['query'];
-    classNameMap['className'] = className;
-    channel.sink.add(jsonEncode(subscribeMessage));
-
-    channel.stream.listen((dynamic message) {
-      final Map<String, dynamic> actionData = jsonDecode(message);
-      if (eventCallbacks.containsKey(actionData['op']))
-        eventCallbacks[actionData['op']](actionData);
-    });
+        if (eventCallbacks.containsKey(actionData['op'])) {
+          if (actionData.containsKey('object')) {
+            final Map<String, dynamic> map = actionData['object'];
+            final String className = map['className'];
+            if (className == '_User') {
+              eventCallbacks[actionData['op']](
+                  ParseUser._getEmptyUser().fromJson(map));
+            } else {
+              eventCallbacks[actionData['op']](
+                  ParseObject(className).fromJson(map));
+            }
+          } else {
+            eventCallbacks[actionData['op']](actionData);
+          }
+        }
+      }, onDone: () {
+        if (_debug) {
+          print('$_printConstLiveQuery: Done');
+        }
+      }, onError: (error, StackTrace stackTrace) {
+        if (_debug) {
+          print('$_printConstLiveQuery: Error: ${error.runtimeType.toString()}');
+        }
+        return Future.value(handleException(error, ParseApiRQ.liveQuery, _debug, _className));
+      });
+
+      //The connect message is sent from a client to the LiveQuery server.
+      //It should be the first message sent from a client after the WebSocket connection is established.
+      _connectMessage = {
+        'op': 'connect',
+        'applicationId': _client.data.applicationId,
+        'clientKey': _client.data.clientKey
```

Comment:
As `clientKey` is optional, this line throws an error if `clientId` is not set in `Parse().initialize( ... )`
```suggestion
        'clientKey': _client.data.clientKey ?? ''
```

Reply:
it is true.  I had not thought of that.  I have to review the whole logic because the key can not be sent empty.

Reply:
I tested setting an empty string as key when initializing Parse, that works just fine. So the fix suggested should do the trick.

Reply:
OK. I'm making the change now.

Reply:
Done @phillwiggins